### PR TITLE
Character inventory cleaning

### PR DESCRIPTION
**Pull request by Pimvgd**
Requires merging #1145 first.

### CODE REVIEW

File: src/com/lilithsthrone/game/inventory/AbstractInventory.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	boolean isEmpty() {
+		return duplicateCounts.isEmpty();
+	}
+
+	public void sort() {
+		if (duplicateCounts.size() < 2) {
+			return;
+		}
+		List<T> weaponsToSort = new ArrayList<>(duplicateCounts.keySet());
```
Comment (CognitiveMist) [inline src/com/lilithsthrone/game/inventory/AbstractInventory.java]: Minor gripe: if this is applicable to items, clothing, and weapons, then this variable shouldn't be called weaponsToSort.

Comment (Innoxia) [reply src/com/lilithsthrone/game/inventory/AbstractInventory.java]: I'll get that changed on my end after merging.

**Comment by Innoxia** [comment]
Thank you!

**Comment by Innoxia** [comment]
I just tested this locally, and trying to identify clothing in your inventory via spending arcane essences throws a null pointer exception.

File: src/com/lilithsthrone/game/inventory/AbstractInventory.java (hunk 1 above)
Comment (Pimvgd) [reply src/com/lilithsthrone/game/inventory/AbstractInventory.java]: Fixed

**Comment by Pimvgd** [comment]
Fixed null pointer; was caused by trying to be too fancy with computeIfPresent ... it's the one bit of code I had trouble with because I wanted to make it smaller and I ended up overcompressing. The new version is a lot more readable.

**Comment by Innoxia** [comment]
Thank you for this! I've merged it locally, and will push to github in about 12 hours, so should show as merged then. ^^

**Pull request merged**